src/components/ActionButton/ActionButton.ts

```diff
@@ -91,6 +91,19 @@ export class ActionButton extends Component {
         }
     }
 
+    public updateIcon(icon: string): void {
```

Comment:
Oops! Forgot headers for added public methods.

Reply:
Why is this method public ?

Reply:
> Why is this method public ?

Because the ActionToggleButton needs to call it. (beware, we're composing, not extending, so no, protected doesn't work)

Reply:
Since _ToggleActionButton_ uses the _ActionButton_ (not extending it), _ToggleActionButton_ can update the icon after the fact. It is the same idea for the tooltip.

We need to be able to use a different icon when the button is activated.

Reply:
Indeed, I'm always warry of new public method 😛